pylint/extensions/typing.py

```diff
@@ -219,6 +224,66 @@ def visit_attribute(self, node: nodes.Attribute) -> None:
         if self._should_check_callable and node.attrname == "Callable":
             self._check_broken_callable(node)
 
+    @only_required_for_messages("redundant-union-assign-typehint")
+    def visit_annassign(self, node: nodes.AnnAssign) -> None:
```

Comment:
At the moment, almost all logic is inside this function. That will make it difficult to reuse for function arguments. Consider moving the logic to a new _private_ helper function.